src/components/PostEditor.js

```diff
@@ -166,20 +171,24 @@ export default class PostEditor extends React.Component {
     })
   }
 
-  findCommunities = term => {
-    if (!term) return
+  updateCommunityChoices = term => {
+    if (!term) {
+      this.setState({communityChoices: []})
+      return
+    }
 
     let { currentUser, postEdit: { communities } } = this.props
     var match = c =>
       startsWith(c.name.toLowerCase(), term.toLowerCase()) &&
       !contains(communities, c.id)
 
-    return filter(currentUser.memberships.map(m => m.community), match)
+    this.setState({communityChoices: filter(currentUser.memberships.map(m => m.community), match)})
   }
 
   render () {
     let { expanded, post, postEdit, dispatch, project } = this.props
     let { name, description, communities, type, location } = postEdit
+    let { communityChoices } = this.state
```

Comment:
The only thing that needs to be in state is the term itself -- you can calculate the communities that correspond to the term in the render method rather than storing them in state

Optionally of course you can then move the term to the app state